grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -283,6 +287,95 @@ class StockMovementController {
         render(template: "documents", model: [stockMovement: stockMovement])
     }
 
+    def comments = {
+        def stockMovement = getStockMovement(params.id)
+        render(template: "comments", model: [stockMovement: stockMovement])
+    }
+
+    def addComment = {
+        def stockMovement = getStockMovement(params.id)
+        List<User> recipients = User.list()
+        return [stockMovement: stockMovement, comment: new Comment(), recipients: recipients]
+    }
+
+    def editComment = {
+        def stockMovement = getStockMovement(params['stockMovement.id'])
+        if (!stockMovement) {
```

Comment:
Is this check required?

Reply:
I don't mind the check, but think the closure should render an error to the page / modal that is being rendered. Same with when the comment is not found.

Reply:
in all cases an error is being rendered.
- if stockMovement doesn't;t exist we redirect to the list page and display an error
- if comment doesn't exist we redirect to stockMovement view page and display an error
- and if user is not a sender then redirect to view page and display an error 

in all cases an error is rendered to a user.